api/node/server/server.go

```diff
@@ -96,15 +98,20 @@ func (s *Server) IntoHandler(mux *http.ServeMux) http.Handler {
 			return response, err
 		}
 	}
-	return HandlerFromMux(NewStrictHandler(s, []StrictMiddlewareFunc{loggingMid}), mux)
-}
-
-func (s *Server) Start(address string) error {
-	server := &http.Server{
-		Handler: s.IntoHandler(http.NewServeMux()),
-		Addr:    address,
+	readinessMid := func(f nethttp.StrictHTTPHandlerFunc, _ string) nethttp.StrictHTTPHandlerFunc {
+		return func(ctx context.Context, w http.ResponseWriter, r *http.Request, req any) (any, error) {
+			select {
+			case <-readiness:
+				return f(ctx, w, r, req)
+			default:
+				// channel is not closed, service is not ready
+				w.WriteHeader(http.StatusServiceUnavailable)
+				return nil, errors.New("service not ready")
+			}
+		}
 	}
```

Comment:
Not a big fan of this approach of passing around the channel. Instead I would prefer to pass the syncer as a dependency to `Server` and call the corresponding method on it here:

```suggestion
	readinessMid := func(f nethttp.StrictHTTPHandlerFunc, _ string) nethttp.StrictHTTPHandlerFunc {
		return func(ctx context.Context, w http.ResponseWriter, r *http.Request, req any) (any, error) {
			if !s.sync.IsSynced(ctx) {
				w.WriteHeader(http.StatusServiceUnavailable)
				return nil, errors.New("service not ready")
			}
			
			return f(ctx, w, r, req)
		}
	}
```

and maybe leave the `Start` method incase it is useful.

Reply:
It's doing something different now because `sync.IsSynced()` checks a different property and it can toggle true/false/true repeatedly. TBH, I'm not sure what the behavior we intend to have is Waiting for the ATX-sync is less restrictive but also more error-prone...

Reply:
> `sync.IsSynced()` checks a different property and it can toggle true/false/true repeatedly.

I don't think this is true. If `sync.IsSynced()` returns `true` it should always return `true` in all subsequent calls. You can check the implementation for it. Same for `RegisterATXSynced()` if it ever returns a closed channel, any subsequent call will also always return a closed channel.

Reply:
`sync.IsSynced()` can switch from `true` to `false` when the node falls behind in syncer layers. Anyway, I'm changing to `syncer.IsSynced()` as we discussed to be the most restrictive about API availability. We can revisit this decision in the future if the API is off (503) too often and relax it then (maybe decide availability per endpoint?).